### ISSUE TITLE
various flaky tests 

### ISSUE THREAD

**Issue by omritoptix** [dymensionxyz/dymint#394]
```
=== RUN   TestMempoolDirectly
I[2023-07-10|12:59:07.553] service start                                module=proxy msg="Starting multiAppConn service" impl=multiAppConn
I[2023-07-10|12:59:07.553] service start                                module=abci-client connection=query msg="Starting localClient service" impl=localClient
I[2023-07-10|12:59:07.554] service start                                module=abci-client connection=snapshot msg="Starting localClient service" impl=localClient
I[2023-07-10|12:59:07.554] service start                                module=abci-client connection=mempool msg="Starting localClient service" impl=localClient
I[2023-07-10|12:59:07.554] service start                                module=abci-client connection=consensus msg="Starting localClient service" impl=localClient
I[2023-07-10|12:59:07.554] service start                                module=events msg="Starting EventBus service" impl=EventBus
I[2023-07-10|12:59:07.554] service start                                module=pubsub msg="Starting PubSub service" impl=PubSub
I[2023-07-10|12:59:07.555] WARNING: working in in-memory mode           
badger 2023/07/10 12:59:07 INFO: Set nextTxnTs to 0
badger 2023/07/10 12:59:07 INFO: Set nextTxnTs to 0
D[2023-07-10|12:59:07.564] Getting latest batch from settlement layer   module=settlement_client 
D[2023-07-10|12:59:07.565] Retrieving batch from settlement layer       module=settlement_client SLstateindex=0
I[2023-07-10|12:59:07.565] Updated latest height from settlement layer  module=settlement_client latestHeight=0
I[2023-07-10|12:59:07.565] Updated sequencers list from settlement layer module=settlement_client sequencersList=[0xc00defc9c0]
I[2023-07-10|12:59:07.565] service start                                module=txindex msg="Starting IndexerService service" impl=IndexerService
I[2023-07-10|12:59:07.566] failed to find state in the store, creating new state from genesis module=BlockManager 
I[2023-07-10|12:59:07.567] service start                                msg="Starting Node service" impl=Node
I[2023-07-10|12:59:07.567] starting P2P client                          
D[2023-07-10|12:59:07.567] starting P2P client                          module=p2p 
I[2023-07-10|12:59:07.582] listening on                                 module=p2p address=/ip4/10.1.0.119/tcp/7676/p2p/12D3KooWSsb6ec3nVvstRLUvJ7WA5ZmnTzJpvNLQfX6Uj9LDvhik
I[2023-07-10|12:59:07.582] listening on                                 module=p2p address=/ip4/127.0.0.1/tcp/7676/p2p/12D3KooWSsb6ec3nVvstRLUvJ7WA5ZmnTzJpvNLQfX6Uj9LDvhik
D[2023-07-10|12:59:07.582] setting up gossiping                         module=p2p 
D[2023-07-10|12:59:07.583] setting up DHT                               module=p2p 
I[2023-07-10|12:59:07.583] no seed nodes - only listening for connections module=p2p 
D[2023-07-10|12:59:07.584] setting up active peer discovery             module=p2p 
D[2023-07-10|12:59:07.585] Mock Data Availability Layer Client starting module=mock 
D[2023-07-10|12:59:07.585] settlement Layer Client starting.            module=settlement_client 
I[2023-07-10|12:59:07.586] started state updates handler loop           module=settlement_client 
I[2023-07-10|12:59:07.586] Starting the block manager                   module=BlockManager 
I[2023-07-10|12:59:07.587] Started retrieve loop                        module=BlockManager 
D[2023-07-10|12:59:07.587] inserted new valid transaction               priority=0 tx=709B55BD3DA0F5A838125BD0EE20C5BFDD7CABA173912D4281CAE816B79A201B height=0 num_txs=1
D[2023-07-10|12:59:07.587] inserted new valid transaction               priority=0 tx=27CA64C092A959C7EDC525ED45E845B1DE6A7590D173FD2FAD9133C8A779A1E3 height=0 num_txs=2
I[2023-07-10|12:59:07.587] Started sync target loop                     module=BlockManager 
D[2023-07-10|12:59:07.588] Getting latest batch from settlement layer   module=settlement_client 
D[2023-07-10|12:59:07.589] Retrieving batch from settlement layer       module=settlement_client SLstateindex=0
E[2023-07-10|12:59:07.589] failed to retrieve batch from SL             module=BlockManager err="batch not found"
D[2023-07-10|12:59:07.688] inserted new valid transaction               priority=0 tx=1F3CB18E896256D7D6BB8C11A6EC71F005C75DE05E39BEAE5D93BBD1E2C8B7A9 height=0 num_txs=3
D[2023-07-10|12:59:07.689] inserted new valid transaction               priority=0 tx=41B637CFD9EB3E2F60F734F9CA44E5C1559C6F481D49D6ED6891F3E9A086AC78 height=0 num_txs=4
D[2023-07-10|12:59:08.488] Gossiping block                              module=p2p len=508
D[2023-07-10|12:59:08.489] Applying block                               module=BlockManager height=4 source=produced
I[2023-07-10|12:59:08.493] block created                                module=BlockManager height=4 num_tx=0
D[2023-07-10|12:59:08.493] indexed block                                module=txindex height=4
D[2023-07-10|12:59:08.493] indexed block txs                            module=txindex height=4 num_txs=0
I[2023-07-10|12:59:08.494] Submitting next batch                        module=BlockManager startHeight=3 endHeight=3 size=512
D[2023-07-10|12:59:08.494] Submitting batch to DA layer                 module=mock startheight=3 endheight=3 daheight=3
D[2023-07-10|12:59:08.495] Submitting batch to settlement layer         module=settlement_client startheight=3 endheight=3
panic: batch start height must be last height. StartHeight 3, lastetHeight 2

```

**Comment by omritoptix** [comment]
it's not just the mempool, it happens also in other tests but the same error:

```
=== RUN   TestHealthStatusEventHandler/TestDAHealthySettlementHealthy
I[2023-07-13|13:43:47.675] All base layers are healthy                  
D[2023-07-13|13:43:47.739] Gossiping block                              module=p2p len=508
D[2023-07-13|13:43:47.740] Applying block                               module=BlockManager height=4 source=produced
D[2023-07-13|13:43:47.745] indexed block                                module=txindex height=4
I[2023-07-13|13:43:47.745] block created                                module=BlockManager height=4 num_tx=0
D[2023-07-13|13:43:47.745] indexed block txs                            module=txindex height=4 num_txs=0
I[2023-07-13|13:43:47.746] Submitting next batch                        module=BlockManager startHeight=3 endHeight=3 size=512
D[2023-07-13|13:43:47.746] Submitting batch to DA layer                 module=mock startheight=3 endheight=3 daheight=7
D[2023-07-13|13:43:47.746] Submitting batch to settlement layer         module=settlement_client startheight=3 endheight=3
panic: batch start height must be last height. StartHeight 3, lastetHeight 2

goroutine 734 [running]:
github.com/dymensionxyz/dymint/settlement.(*BaseLayerClient).SubmitBatch(0xc00147a0f0, 0xc00110adc0, {0x28df767, 0x4}, 0x1fd8174?)
	/home/runner/work/dymint/dymint/settlement/base.go:101 +0x299
github.com/dymensionxyz/dymint/block.(*Manager).submitNextBatch(0xc00141a2c0, {0xa?, 0x1?})
	/home/runner/work/dymint/dymint/block/manager.go:752 +0x78b
created by github.com/dymensionxyz/dymint/block.(*Manager).produceBlock
	/home/runner/work/dymint/dymint/block/manager.go:722 +0x11ea
FAIL	github.com/dymensionxyz/dymint/node	5.209s
```

**Comment by omritoptix** [comment]
closed by #409
